web/lumpTypeView.ts

```diff
@@ -572,3 +578,169 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
         }
     });
 }
+
+export function LumpTypeViewColormapAll(scaleX:number = 2, scaleY:number = 2) {
+    return new LumpTypeView({
+        name: "Colormap (all)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            const cmap = lumps.WADColorMap.from(lump);
+
+            // Get PLAYPAL, or use default if it isn't found.
+            const files = getWadFileList(lump);
+            const playpal = files.getPlaypal();
+
+            // Set up canvas
+            const displayWidth = playpal.getColorPerPaletteCount() * scaleX;
+            const displayHeight = cmap.getMapCount() * scaleY;
+            const canvas = util.createElement({
+                tag: "canvas",
+                class: "lump-view-image",
+                appendTo: root
+            });
+            canvas.width = displayWidth;
+            canvas.height = displayHeight;
+
+            const context = canvas.getContext("2d") as CanvasRenderingContext2D;
+            const data = cmap.getPixelDataRGBA(playpal, 0);
+
+            // Draw the colormap
+            for (let cmapIdx = 0; cmapIdx < cmap.getMapCount(); cmapIdx++) {
+                for (let palIdx = 0; palIdx < playpal.getColorPerPaletteCount(); palIdx++) {
+
+                    // Figure out where to paint
+                    const row = cmapIdx * scaleY;
+                    const col = palIdx * scaleX;
+                    const width = scaleX;
+                    const height = scaleY;
+
+                    // Data index
+                    const didx = (palIdx + cmapIdx * playpal.getColorPerPaletteCount()) * 4;
+
+                    // Get RGB string
+                    let rgb = data.readUInt32BE(didx).toString(16);
+                    while (rgb.length < 8) {
+                        rgb = "0" + rgb;
+                    }
+
+                    // Set style and fill colour
+                    context.fillStyle = `#${rgb.substring(0, 6)}`;
+                    context.fillRect(col, row, width, height);
+                }
+            }
+        }
+    });
+}
+
+export function LumpTypeViewColormapByMap(scaleX:number = 4, scaleY:number = 4) {
```

Comment:
Style guide nitpick: Please go through the variable names in this function and make them easier for a newcomer to read and understand? i.e. fewer abbreviations and more descriptive names.

imo `num` abbreviating `number of` is fine. Perhaps `numCmaps` could become `numColormaps`?

Reply:
I don't like `cur` abbreviating `current`, though, or `el` abbreviating `element`, and I really don't like `idx` abbreviating `index`. (You're only saving two characters!?)

`container` and `container2` could perhaps become `innerContainer` and `outerContainer`, respectively?